Documentation/Assertions.rst

```diff
@@ -0,0 +1,72 @@
+.. include:: /Includes.rst.txt
+
+.. _assertions-and-type-checks:
+
+==========================================
+Our approach to assertions and type checks
+==========================================
```

Comment:
```suggestion
=========================================
Our approach to assertions and type checks
=========================================
```

Reply:
I'm confused … on my screen, your change makes the `===` lines 1 char shorter than the text line, and AFAIK the `===` lines should be the same length. Am I missing something here?

Reply:
They should. Looks like an issue with GitHub on my end than.
https://raw.githubusercontent.com/FriendsOfTYPO3/tea/3949ed3ea7cfc4ed58ebfb94a2be1c41889e97a4/Documentation/Assertions.rst looks okay.

Screenshot for reference:
![image](https://github.com/FriendsOfTYPO3/tea/assets/354250/c5a7c88c-187e-40ba-81a0-c1c24db773e5)